src/carousel.js

```diff
@@ -262,21 +263,37 @@ class Carousel extends React.Component {
     this.autoSlide('prev')
   }
 
+  setFrame (index, ms = 450) {
+    const diff = Math.abs(index - this.state.currentFrameIndex);
+    if (index < this.state.currentFrameIndex) {
+      for (let i = 0; i < diff; i++) {
+        setTimeout(() => this.prev(), i * ms);
```

Comment:
There would be a series of animation for switching to target frame, which works but not ideal, especially when user jumps between a lot frames. We need to switch to target frame directly. 🤔

Reply:
Ok I preferred to have the animation and to be able to control the animation time. For a small number of frames setting 'ms' to 0 will visually have the same effect, but for many frames this could be an issue as you say. But I will look into creating a function that allows changing it directly.

Reply:
@amio I have updated the PR. Also updated index.js so that the example is using this functionality. Currently setFrameIndex is set to 4 to show how it works, but it could be removed.